sls-versions/src/main/java/com/palantir/sls/versions/SlsVersionMatcher.java

```diff
@@ -70,38 +71,32 @@ public static SlsVersionMatcher valueOf(String value) {
     /** The same as {@link #valueOf(String)}, but returns {@link Optional#empty} if the format is invalid. */
     public static Optional<SlsVersionMatcher> safeValueOf(String value) {
         checkNotNull(value, "value cannot be null");
+        return SlsVersionMatcherParser.safeValueOf(value);
+    }
 
-        Matcher matcher = PATTERN.matcher(value);
-        if (!matcher.matches()) {
+    static Optional<SlsVersionMatcher> maybeCreate(
+            String value, OptionalInt major, OptionalInt minor, OptionalInt patch) {
+        SlsVersionMatcher maybeMatcher = ImmutableSlsVersionMatcher.of(value, major, minor, patch);
```

Comment:
this line did change though - it used to be a builder and now it's a @Value.Parameter single-shot construct-it-all invocation.